core/memory/arena/cc/arena.cpp

```diff
@@ -184,12 +188,10 @@ void* Arena::allocate(uint32_t size, uint32_t align) {
       // The first allocation is offset by 1 block.
       allocation = data.current_chunk + block_size;
       data.offset_of_next_allocation_in_chunk = 2 * block_size;
+      header = reinterpret_cast<chunk_header*>(data.current_chunk);
     }
+    header->num_allocations += 1;
```

Comment:
It looks like the `num_allocations` are now always guarded by the lock - does it still need to be atomic?